docs/how-to-guides/use-a-custom-timestamp-column.md

```diff
@@ -0,0 +1,213 @@
++------------------------------------+------------------------------------+------------------------------------+------------------------------------+------------------------------------+------------------------------------+
+|k1                                  |1589564380616                       |2020-05-15 17:39:40.616             |2020-05-04 01:00:00                 |0                                   |a                                   |
+|k2                                  |1589564380731                       |2020-05-15 17:39:40.731             |2020-05-04 02:00:00                 |1                                   |b                                   |
+```
+
+Derive a new stream, `s2`, from `s1` and tell ksqlDB to use event-time. Set the `timestamp` property to the `ts` column.
+
+```sql
+CREATE STREAM S2 WITH (
+    timestamp = 'ts',
+    timestamp_format = 'yyyy-MM-dd HH:mm:ss'
+)   AS
+    SELECT *
+    FROM s1
+    EMIT CHANGES;
+```
+
+Now compare the timestamps again. This time, notice that `ROWTIME` has been set to the same value as `ts`. `s2` is now using event-time.
+
+```sql
+SELECT k,
+       ROWTIME,
+       TIMESTAMPTOSTRING(ROWTIME, 'yyyy-MM-dd HH:mm:ss.SSS') AS rowtime_formatted,
+       ts,
+       v1,
+       v2
+FROM s2
+EMIT CHANGES;
+```
+
+The query should return the following results.
+
+```
++------------------------------------+------------------------------------+------------------------------------+------------------------------------+------------------------------------+------------------------------------+
+|K                                   |ROWTIME                             |ROWTIME_FORMATTED                   |TS                                  |V1                                  |V2                                  |
++------------------------------------+------------------------------------+------------------------------------+------------------------------------+------------------------------------+------------------------------------+
+|k1                                  |1588554000000                       |2020-05-04 01:00:00.000             |2020-05-04 01:00:00                 |0                                   |a                                   |
+|k2                                  |1588557600000                       |2020-05-04 02:00:00.000             |2020-05-04 02:00:00                 |1                                   |b                                   |
+```
+
+Any new streams or tables derived from `s2` will continue to have their timestamp set to `ts` unless an operation instructs otherwise.
+
+
+## Timestamps on base streams/tables
+
+Not only can you change the timestamp to use as you derive new streams and tables, you can also set it on base ones, too. Simply set the `timestamp` and `timestamp_format` properties on the `WITH` clause.
```

Comment:
we might want to note that this won't change the data in the stream, it just instructs all downstream queries to use the timestamp field instead of the rowtime

Reply:
When you say that it doesn't change the data in the stream, what you mean is that the metadata for the underlying Kafka records are not changed, right?

My understanding is that from a users point of view, as soon as you interact with the stream in any way (select whatever), rowtime is set to the indicated timestamp.